### PR TITLE
Feature: Add CI

### PR DESCRIPTION
**Pull request by infeo**
This PR adds CI to the project by using Github Actions.

Two workflows exists:
* build: to build and test the JAR. It is triggered on push/tag. If the github ref is a tag, then a release is drafted. The workflow uploads reports like dependency-check or surefire
* native-image: to build a native image with graal VM. Triggerd manually or on a published release. In the latter case, the image is build with higher optimization. Afterwards, the executable and the autocompletion file is uploaded.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -19,9 +19,32 @@ jobs:
...
+        if: startsWith(github.ref, 'refs/tags/')
+        uses: softprops/action-gh-release@v1
+        with:
+          draft: true
+          token: ${{ secrets.CRYPTOBOT_RELEASE_TOKEN }}
+          generate_release_notes: true
+          body: |-
+            :construction: Work in Progress
+            
+            ⏳ Please be patient, the builds are still [running](https://github.com/cryptomator/cryptomator/actions). New versions of Cryptomator can be found here in a few moments. ⏳
```
Comment (overheadhunter) [inline .github/workflows/build.yml]: wrong url

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -19,9 +19,32 @@ jobs:
...
+          cache: 'maven'
           github-token: ${{ secrets.GITHUB_TOKEN }}
       - name: Build and Test
         run: >
-          mvn -B clean package -Pjar
+          mvn -B clean verify -Pdependency-check
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Needed to get PR information, if any
\ No newline at end of file
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Needed to get PR information, if any
```
Comment (overheadhunter) [inline .github/workflows/build.yml]: only required for SonarCloud

File: .github/workflows/native-image.yml [hunk 3]
```diff
@@ -12,17 +16,32 @@ jobs:
...
           distribution: 'graalvm'
+          cache: 'maven'
           github-token: ${{ secrets.GITHUB_TOKEN }}
-      - name: Build native image
+      - name: Build develop native image
+        if: ${{ !startsWith(github.ref, 'refs/tags/') }}
         run: >
           mvn -B clean package -Pnative
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Needed to get PR information, if any
```
Comment (overheadhunter) [inline .github/workflows/native-image.yml]: only required for SonarCloud

**Review by overheadhunter — CHANGES_REQUESTED** [review]

File: .github/workflows/native-image.yml (hunk 3 above)
Comment (infeo) [reply .github/workflows/native-image.yml]: removed in https://github.com/cryptomator/hub-cli/pull/5/commits/370686fd2594bf8b97753bca28b90c616df79140

File: .github/workflows/build.yml (hunk 2 above)
Comment (infeo) [reply .github/workflows/build.yml]: removed in https://github.com/cryptomator/hub-cli/pull/5/commits/370686fd2594bf8b97753bca28b90c616df79140

**Review by overheadhunter — APPROVED** [review]

**Pull request merged**